media/js/newsletter/form-utils.es6.js

```diff
@@ -14,6 +14,91 @@ function checkEmailValidity(email) {
     return /\S+@\S+/.test(email);
 }
 
+/**
+ * Hide all visible form error labels.
+ * @param {HTMLFormElement} form
+ */
+function clearFormErrors(form) {
+    const errorMsgs = form.querySelectorAll('.mzp-c-form-errors li');
+
+    form.querySelector('.mzp-c-form-errors').classList.add('hidden');
+
+    for (let i = 0; i < errorMsgs.length; i++) {
+        errorMsgs[i].classList.add('hidden');
+    }
```

Comment:
Not a criticism, more of a curiosity, but why not use something like:

```
errorMsgs.map((e) => e.classList.add('hidden'));
```

Reply:
Fair call out!

The honest answer here is much of this code is repurposed from the existing newsletter code. If I was to think a little more about it though, I'm not sure your example would work out of the box, as we'd need to convert the node list to an array first. So it might be something more like:

```javascript
Array.from(errorMsgs).map((e) => e.classList.add('hidden'));
```

Maybe I'm an old timer here, and whilst this is certainly less code (only one line!), I'm not convinced it's necessarily easier to read or follow. I also suspect the conversion is going to be slower performance wise (although I admittedly have not checked).

Reply:
I was curious so I put together a little comparison here. The for loop is about 15.17% faster: https://jsbench.me/2wl8y9vkvl/1

Reply:
We were chatting about this in the front-enders channel a bit, and another option (which is definitely more readable than a `for` loop) could have been [forEach](https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach) which works on node lists, but it doesn't appear to be part of an official spec (according to MDN)